### PR TITLE
Fixed incorrect handling of a weekly RRULE at the beginning of the year

### PR DESCRIPTION
**Pull request by ksuther**
Found an edge case for weekly recurrences at the beginning of the year. Regression test included.

---

This RRULE on January 1 would result in the second occurrence being January 7 rather than January 2.

FREQ=WEEKLY;WKST=SU;INTERVAL=2;BYDAY=MO,TU,WE,TH,FR

The problem is that day_diff wasn't handling cases where a > b and would count days of the year incorrectly. Temporarily swaping a and b and then performing the day_diff calculation avoids the issue.

### CODE REVIEW

File: src/libical/icalrecur.c [hunk 1]
```diff
@@ -1959,13 +1959,29 @@ static int __day_diff(icalrecur_iterator *impl, icaltimetype a, icaltimetype b)
         diff = get_day_of_year(impl, b.year, b.month, b.day, NULL) -
             get_day_of_year(impl, a.year, a.month, a.day, NULL);
     } else {
+        /* Swap a and b if a is greater than b */
+        int flipped = 0;
```
Comment (winterz) [inline src/libical/icalrecur.c]: in order to make MSVC happy, you probably have to move the declaration for 'int flipped' near the top of the function.

**Comment by ksmurchison** [comment]
Good catch Kent!  This fix works for me.

**Comment by ksuther** [comment]
I think `flipped` is fine, it's that `year` ended up getting pushed down. I'll see if that builds on MSVC 2012 now.

**Comment by ksuther** [comment]
Should be better now.
